### PR TITLE
feat - #175 adds processSource option to the loader config to run han…

### PR DESCRIPTION
**Pull request by pizatski**
…dlebars template through html minifier

### CODE REVIEW

**Comment by pizatski** [comment]
Looks like I auto-formatted a bunch of stuff. Let me know if that will be an issue. It passed the linter configured.

Also, I'm addressing the Hound violation.

File: index.js [hunk 1]
```diff
@@ -6,6 +6,7 @@ var path = require("path");
 var assign = require("object-assign");
 var fastreplace = require('./lib/fastreplace');
 var findNestedRequires = require('./lib/findNestedRequires');
+var minify = require('html-minifier').minify;
```
Comment (pizatski) [inline index.js]: Yes, this is an added dependency. I feel like it's probably better to add a known and tested library to do this sort of thing than writing my own regular expressions which will surely be buggy.

File: index.js [hunk 2]
```diff
@@ -26,7 +27,23 @@ function getLoaderConfig(loaderContext) {
...
+ *
+ * @param {String} source
+ * @returns {String}
+ */
+var processSource = function (source) {
+  return minify(source, {
+    collapseWhitespace: true,
+    conservativeCollapse: true,
+    removeComments: true,
+    removeAttributeQuotes: true
```
Comment (pizatski) [inline index.js]: These are fairly conservative options. More can be found here:
https://www.npmjs.com/package/html-minifier

File: index.js [hunk 3]
```diff
@@ -202,14 +217,17 @@ module.exports = function(source) {
 
     try {
       if (source) {
+        if (query.processSource) {
```
Comment (pizatski) [inline index.js]: I got a lint error when I double banged this. Let me know if that's okay.

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,11 @@
 ### Fixed
 - Your improvement here...
 
+## [1.7.2] - 2019-01-09
+
+### Fixed
```
Comment (pcardune) [inline CHANGELOG.md]:
```suggestion
### New Features
```

File: README.md [hunk 5]
```diff
@@ -64,6 +64,7 @@ The following query (or config) options are supported:
  - *ignorePartials*: Prevents partial references from being fetched and bundled. Useful for manually loading partials at runtime.
  - *ignoreHelpers*: Prevents helper references from being fetched and bundled. Useful for manually loading helpers at runtime.
  - *precompileOptions*: Options passed to handlebars precompile. See the Handlebars.js [documentation](http://handlebarsjs.com/reference.html#base-precompile) for more information.
+ - *processSource*: Runs the handlebars templates through an html minifier to remove whitespace and reduce the size of the compiled templates.
```
Comment (pcardune) [inline README.md]: `processSource` is a very generic name that makes it hard to guess how exactly the source is being processed. Maybe something like `minifyOutput` would be better?

Comment (pizatski) [reply README.md]: I think `minifyOutput` is a much clearer name. Will make the change.

**Comment by pizatski** [comment]
Closing this pull request. I've come to agree that this library shouldn't really be responsible for handling magnification unless there is much more traction put under the feature request from the community.

**Pull request closed without merging**